File: instagram-backend/src/main/java/com/hoanghuy04/instagrambackend/entity/Comment.java
package com.hoanghuy04.instagrambackend.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DocumentReference;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * Entity representing a comment on a post or reply to another comment.
 * Supports nested replies through self-referencing.
 * 
 * @author Instagram Backend Team
 * @version 1.0.0
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "comments")
public class Comment {

    @Id
    private String id;

    @DocumentReference
    private Post post;

    @DocumentReference
    private User author;

    private String text;

    @DocumentReference
    private Comment parentComment;

    @Builder.Default
    private long totalLikes = 0;

    private String mention;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: instagram-frontend-mobile/types/comment.ts
import { UserSummaryResponse } from './user';

export interface Comment {
  id: string;
  postId: string;
  parentCommentId?: string | null;
  author: UserSummaryResponse;
  text: string;
  mention?: string;
  likesCount: number;
  repliesCount?: number;
  isLiked?: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCommentRequest {
  postId: string;
  text: string;
  mention?: string;
}

export interface UpdateCommentRequest {
  text: string;
}
